myriad-scheduler/src/main/java/com/ebay/myriad/api/ClustersResource.java

```diff
@@ -100,46 +95,114 @@ public Response flexUp(FlexUpClusterRequest request) {
+
+        Integer numFlexedUp = this.getNumFlexedupNMs();
+        if (isValidRequest && numFlexedUp < instances)  {
+            String message = String.format("Number of requested instances for flexdown is greater than the number of " +
+                "Node Managers previously flexed up. Requested: %d, Previously flexed Up: %d. " +
+                "Only %d Node Managers will be flexed down", instances, numFlexedUp, numFlexedUp);
+            response.entity(message);
+            LOGGER.warn(message);
         }
 
         Response returnResponse = response.build();
         if (returnResponse.getStatus() == Response.Status.ACCEPTED.getStatusCode()) {
-            this.myriadOperations.flexDownCluster(instances);
+            String constraint = constraints != null && !constraints.isEmpty() ? constraints.get(0) : null;
+            this.myriadOperations.flexDownCluster(profileManager.get(profile),
+                ConstraintFactory.createConstraint(constraint), instances);
         }
         return returnResponse;
     }
 
-    private boolean isValidInstanceSize(Integer instances) {
-        return (instances > 0);
+    private boolean validateProfile(String profile, ResponseBuilder response) {
+      if (profile == null || profile.isEmpty()) {
+        response.status(Response.Status.BAD_REQUEST).entity("'profile' is null or empty");
+        LOGGER.error("'profile' is null or empty");
+        return false;
+      }
+      if (!this.profileManager.exists(profile)) {
+        response.status(Response.Status.BAD_REQUEST)
+            .entity("Profile does not exist: '" + profile + "'");
+        LOGGER.error("Provided profile does not exist: '" + profile + "'");
+        return false;
+      }
+      return true;
+    }
+
+    private boolean validateInstances(Integer instances, ResponseBuilder response) {
+      if (instances == null) {
+        response.status(Response.Status.BAD_REQUEST).entity("'instances' is null");
+        LOGGER.error("'instances' is null");
+        return false;
+      } else if (!(instances > 0)) {
+          response.status(Response.Status.BAD_REQUEST)
+                  .entity("Invalid instance size: " + instances);
+          LOGGER.error("Invalid instance size request " + instances);
+        return false;
+      }
+      return true;
+    }
+
+    private boolean validateConstraints(List<String> constraints, ResponseBuilder response) {
+      if (constraints != null && !constraints.isEmpty()) {
+        boolean valid = validateConstraintsSize(constraints, response);
+        valid = valid && validateLIKEConstraint(constraints.get(0), response);
+        return valid;
+      }
+      return true;
     }
 
-    // TODO (mohit): put this in Myriad Operations?
-    private Integer getFlexibleInstances() {
-        // this follows the logic of myriadOperations.flexDownCluster
+    private boolean validateConstraintsSize(List<String> constraints, ResponseBuilder response) {
+      if (constraints.size() > 1) {
```

Comment:
If later on we will have multiple constraints how will we define relationship between them: "OR", "AND"?